src/schema/submissions.ts

```diff
@@ -132,21 +138,28 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         .getRepository(User)
         .findOneByOrFail({ id: ctx.userId });
 
-      if (!hasSubmissionAccess(user)) {
-        return {
-          result: 'rejected',
-          reason: SubmissionFailErrorMessage.ACCESS_DENIED,
-        };
-      }
+      const isTeamMember = await ctx.con
+        .getRepository(Feature)
+        .findOneBy({ feature: FeatureType.Team, userId: ctx.userId });
 
       const submissionRepo = ctx.con.getRepository(Submission);
-      const submissionsToday = await getSubmissionsToday(ctx.con, user);
 
-      if (submissionsToday.length >= submissionLimit) {
-        return {
-          result: 'rejected',
-          reason: SubmissionFailErrorMessage.LIMIT_REACHED,
-        };
+      if (!isTeamMember) {
+        if (!hasSubmissionAccess(user)) {
+          return {
+            result: 'rejected',
+            reason: SubmissionFailErrorMessage.ACCESS_DENIED,
+          };
+        }
```

Comment:
What if instead of nested if we have `const limit = isTeamMember ? Infinity : submissionLimit`, the rest of the logic would be the same?

Also I think we should still check `hasSubmissionAccess` even if user is a team member.

Reply:
The check just happens on 2 sides making it a bit more tricky and it's more of a hack at the moment.
I'll check though if I can easily do it :)